backend/src/main/java/com/google/minicrm/data/Settings.java

```diff
@@ -47,10 +87,27 @@ public Settings(Entity entity) {
         this.phone = (String) entity.getProperty("phone");
         this.phoneNotificationsFrequency =
                 NotificationsFrequency.fromDisplayed((String) entity.getProperty("phoneNf"));
-        this.language = Language.fromIsoCode((String) entity.getProperty("language"));
         this.currency = Currency.fromIsoCode((String) entity.getProperty("currency"));
     }
 
+    public static Settings fromReader(Reader reader, Key advertiserKey) throws NumberParseException, ValidatorException, MalformedJsonException {
+        Gson gson = new Gson();
+        Map<String, String> jsonMap = gson.fromJson(reader, Map.class);
+        if(jsonMap == null || !jsonMap.containsKey("email") || !jsonMap.containsKey("phone") ||
```

Comment:
alternatively, maybe just convert these "! containsKey" checks to "&&"and just put a "!" on the entire statement, to reduce number of "!" required.